contracts/contracts/strategies/balancer/BaseBalancerStrategy.sol

```diff
@@ -0,0 +1,323 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title OETH Base Balancer Abstract Strategy
+ * @author Origin Protocol Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20, InitializableAbstractStrategy } from "../../utils/InitializableAbstractStrategy.sol";
+import { IBalancerVault } from "../../interfaces/balancer/IBalancerVault.sol";
+import { IRateProvider } from "../../interfaces/balancer/IRateProvider.sol";
+import { VaultReentrancyLib } from "./VaultReentrancyLib.sol";
+import { IOracle } from "../../interfaces/IOracle.sol";
+import { IVault } from "../../interfaces/IVault.sol";
+import { IWstETH } from "../../interfaces/IWstETH.sol";
+import { IERC4626 } from "../../../lib/openzeppelin/interfaces/IERC4626.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+import "hardhat/console.sol";
+
+abstract contract BaseBalancerStrategy is InitializableAbstractStrategy {
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+    IBalancerVault internal immutable balancerVault =
```

Comment:
I think this immutable should be set in the constructor so unit tests can mock the balancer vault.

Reply:
Yup good point